tests/api/mam_test.c

```diff
@@ -136,6 +136,50 @@ void test_write_until_next_channel(void) {
   }
 }
 
+void test_write_with_chid(void) {
+  const int beginning_msg_num = 1;
+  char seed[NUM_TRYTES_ADDRESS + 1] = {};
+  const char* json_template_send =
+      "{\"x-api-key\":\"" TEST_TOKEN
+      "\",\"data\":{\"seed\":\"%s\",\"ch_mss_depth\":1,\"message\":\"%s:%d\"}, \"protocol\":\"MAM_V1\"}";
+  const char payload[] = "This is test payload number";
+  const int len = strlen(json_template_send) + NUM_TRYTES_ADDRESS + strlen(payload) + 2;
+  gen_rand_trytes(NUM_TRYTES_ADDRESS, (tryte_t*)seed);
+  double sum = 0;
+  ta_send_mam_res_t* res;
+  test_time_start(&start_time);
+  char* json_result = NULL;
+  for (int i = 0; i < beginning_msg_num; ++i) {
+    res = send_mam_res_new();
+    char* json = (char*)malloc(sizeof(char) * len);
+    snprintf(json, len, json_template_send, seed, payload, i);
+    TEST_ASSERT_EQUAL_INT32(
+        SC_OK, api_send_mam_message(&ta_core.ta_conf, &ta_core.iota_conf, &ta_core.iota_service, json, &json_result));
+    free(json);
+    if (i != beginning_msg_num - 1) {
+      free(json_result);
+      send_mam_res_free(&res);
+    }
+  }
+  send_mam_res_deserialize(json_result, res);
+  free(json_result);
+
+  // Send message from next channel ID
+  const char* json_template_send_chid =
+      "{\"x-api-key\":\"" TEST_TOKEN
+      "\",\"data\":{\"seed\":\"%s\",\"chid\":\"%s\",\"ch_mss_depth\":2,\"message\":\"%s\"}, \"protocol\":\"MAM_V1\"}";
+  const int len_send_chid = strlen(json_template_send_chid) + NUM_TRYTES_ADDRESS * 2 + strlen(payload);
+  char* json_send_chid = (char*)malloc(sizeof(char) * (len_send_chid + 1));
+  snprintf(json_send_chid, len_send_chid, json_template_send_chid, seed, res->chid1, payload);
+  TEST_ASSERT_EQUAL_INT32(SC_OK, api_send_mam_message(&ta_core.ta_conf, &ta_core.iota_conf, &ta_core.iota_service,
+                                                      json_send_chid, &json_result));
+  test_time_end(&start_time, &end_time, &sum);
+  send_mam_res_free(&res);
+  free(json_send_chid);
+  free(json_result);
+  printf("Time of write_with_chid: %lf\n", sum);
```

Comment:
Should be "elapsed time"

Reply:
Does it make sense to call `printf` rather than logger functions?

Reply:
We use `printf` to print testing messages in `/tests`